### PR TITLE
DATACOUCH-329 Save repo return with the generated id

### PR DESCRIPTION
**Pull request by MLabusquiere**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATACOUCH).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@MLabusquiere Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-couchbase&pullRequestId=151)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-couchbase&pullRequestId=151) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@MLabusquiere Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-couchbase&pullRequestId=151)!

File: src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java [hunk 1]
```diff
@@ -586,6 +586,7 @@ private void doPersist(Object objectToPersist, final PersistTo persistTo, final
     execute(new BucketCallback<Boolean>() {
       @Override
       public Boolean doInBucket() throws InterruptedException, ExecutionException {
+        String generatedId = addCommonPrefixAndSuffix(converted.getId());
         converted.setId(addCommonPrefixAndSuffix(converted.getId()));
```
Comment (simonbasle) [inline src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java]: reuse the `generatedId` variable here?

**Comment by MLabusquiere** [comment]
Thanks @subalakr. May close the pull request?
Do you have any idea when  2.2.X branch will be released?

**Comment by subalakr** [comment]
Thanks much for the contribution. I am not sure about 2.2.x release. I've ported the feature to 2.3.x branch and the next service release date for that branch is September 4th.

**Pull request closed without merging**